sdk/src/compile.rs

```diff
@@ -0,0 +1,202 @@
+use std::fmt::Display;
+use std::fs;
+use std::io;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::Command;
+use std::str::FromStr;
+use uuid::Uuid;
+
+pub use crate::error::BuildError;
+
+#[doc(hidden)]
+#[derive(Default)]
+pub enum ForProver {
+    #[default]
+    Default,
+    Jolt,
+}
+
+impl Display for ForProver {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Default => write!(f, "default"),
+            Self::Jolt => write!(f, "jolt"),
+        }
+    }
+}
+
+/// Options for dynamic compilation of guest programs.
+///
+/// By default, compilation occurs within `/tmp`. However, the implementation does respect the [`OUT_DIR`](https://doc.rust-lang.org/cargo/reference/environment-variables.html) environment variable.
+#[derive(Clone)]
+pub struct CompileOpts {
+    /// The (in-workspace) package to build.
+    pub package: String,
+    /// The binary produced by the build that should be loaded into the zkVM after successful compilation.
+    pub binary: String,
+    debug: bool,
+    //native: bool,
+    unique: bool,
+    pub(crate) memlimit: Option<usize>, // in mb
+}
+
+impl CompileOpts {
+    /// Setup options for dynamic compilation.
+    pub fn new(package: &str) -> Self {
+        Self {
+            package: package.to_string(),
+            binary: package.to_string(),
```

Comment:
a note that it's not a big change to detect binary name from cargo manifest, as the error returned to user is a unit enum without a message.

Can be implemented in 0.2.1